cloud/shared/bin/lib/terraform.sh

```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env bash
+
+#######################################
+# Generates terraform variable files and runs terraform init and apply.
+# Also initializes the storage bucket for tfstate if it's not setup yet.
+# Globals:
+#   TERRAFORM_TEMPLATE_DIR
+#   BACKEND_VARS_FILENAME
+#   TF_VAR_FILE
+#######################################
+function terraform::perform_apply() {
```

Comment:
i think we can use this in the setup script as well

Reply:
`cloud/azure/bin/setup_tf_shared_state` relies on a bunch of bash functions being loaded first, any thoughts on how that would work with calling this from py?

Reply:
wait sorry does it not source the things it needs? also we ALSo probably need a way to handle if the user doesn't want to use backend vars. i guess like can't we do the like source xyz && call the script right?

Reply:
> wait sorry does it not source the things it needs?

it does now

Reply:
> also we ALSo probably need a way to handle if the user doesn't want to use backend vars

what var do we switch on for that?

Reply:
done